### PR TITLE
Refactor away from food as medicine

### PR DESCRIPTION
**Pull request by Lev-Stambler**
- Remove `impact` and `recommendation` system and replace with `query`
- Rename the scholars parser to the more specific Europe PMC parser to allow for different databases

### CODE REVIEW

**Review by vitashubin — COMMENTED** [review]
Does this mean queries such as "cabbage for acne" will return different scores than "cabbage acne"? Do you do any trimming of the query string?

**Comment by Lev-Stambler** [comment]
> Does this mean queries such as "cabbage for acne" will return different scores than "cabbage acne"? Do you do any trimming of the query string?

I do not. That is a good idea though, I will add it to the p2 goals

**Comment by vitashubin** [comment]
> > Does this mean queries such as "cabbage for acne" will return different scores than "cabbage acne"? Do you do any trimming of the query string?
> 
> I do not. That is a good idea though, I will add it to the p2 goals

Hmm I'm a little worried about this just because the exact string "cabbage for acne" might not show up in texts, but "cabbage can be good when treating acne" for example might still show up. in that sense I think it was better when you counted the two words separately, can we at least split the string and search for key terms individually?

**Comment by Lev-Stambler** [comment]
> > > Does this mean queries such as "cabbage for acne" will return different scores than "cabbage acne"? Do you do any trimming of the query string?
> > 
> > 
> > I do not. That is a good idea though, I will add it to the p2 goals
> 
> Hmm I'm a little worried about this just because the exact string "cabbage for acne" might not show up in texts, but "cabbage can be good when treating acne" for example might still show up. in that sense I think it was better when you counted the two words separately, can we at least split the string and search for key terms individually?

I agree. I think I could add a filter which removes filler words. (Words in the query are already stemmed downstream in the correlation-score algorithm)

**Comment by vitashubin** [comment]
> > > > Does this mean queries such as "cabbage for acne" will return different scores than "cabbage acne"? Do you do any trimming of the query string?
> > > 
> > > 
> > > I do not. That is a good idea though, I will add it to the p2 goals
> > 
> > 
> > Hmm I'm a little worried about this just because the exact string "cabbage for acne" might not show up in texts, but "cabbage can be good when treating acne" for example might still show up. in that sense I think it was better when you counted the two words separately, can we at least split the string and search for key terms individually?
> 
> I agree. I think I could add a filter which removes filler words. (Words in the query are already stemmed downstream in the correlation-score algorithm)

Great, lets file a bug to track that before submitting this PR, otherwise LGTM :)

File: libs/article-parser/src/lib/correlation-score/correlation-score.ts [hunk 1]
```diff
@@ -78,29 +60,19 @@ function stemString(input: string) {
...
-  impacted: string,
-  recommendation: string,
-  impactedSynonyms: string[],
-  recommendationSynonyms: string[]
+  query: string,
+  querySynonyms: string[]
 ): ParsedArticleParagraph {
-  const impactedStem = stemString(impacted);
+  const queryStem = stemString(query);
   const paragraphStemmed = stemString(paragraph);
```
Comment (vitashubin) [inline libs/article-parser/src/lib/correlation-score/correlation-score.ts]: can we be consistent with the naming here? (one is named xStem, and one is xStemmed)

**Review by vitashubin — APPROVED** [review]

**Pull request merged**